lib/parsers/moov_parser.rb

```diff
@@ -115,5 +118,33 @@ def matches_moov_definition?(io)
     maybe_atom_size >= minimum_ftyp_atom_size && maybe_ftyp_atom_signature == 'ftyp'
   end
 
+  def parse_sample_atom(decoder, atom_tree)
+    video_trak_atom = decoder.find_video_trak_atom(atom_tree)
+
+    stts = begin
+      if video_trak_atom
+        decoder.find_first_atom_by_path([video_trak_atom], 'trak', 'mdia', 'minf', 'stbl', 'stts')
+      else
+        decoder.find_first_atom_by_path(atom_tree, 'moov', 'trak', 'mdia', 'minf', 'stbl', 'stts')
+      end
+    end
+
+    mdhd = begin
+      if video_trak_atom
+        decoder.find_first_atom_by_path([video_trak_atom], 'trak', 'mdia', 'mdhd')
+      else
+        decoder.find_first_atom_by_path(atom_tree, 'moov', 'trak', 'mdia', 'mdhd')
+      end
+    end
+
+    if stts && mdhd
+      timescale = mdhd.atom_fields[:tscale]
+      sample_duration = stts.field_value(:entries).first[:sample_duration]
+      return timescale.to_f / sample_duration
```

Comment:
Are we sure this won't raise any division error? I see a similar operation above, but I wanted to double check.

Reply:
those fields must be there if the whole atom is there. However a file can be corrupted or not valid so this is a point too and added extra checks to make sure we are not doing illegal operations